arraycontext/impl/pytato/__init__.py

```diff
@@ -232,77 +251,89 @@ def __init__(self, queue: "cl.CommandQueue", allocator=None,
         # unused, but necessary to keep the context alive
         self.context = self.queue.context
 
-    # {{{ ArrayContext interface
-
-    def clone(self):
-        return type(self)(self.queue, self.allocator)
+    @property
+    def frozen_array_types(self) -> Tuple[Type, ...]:
+        import pyopencl.array as cla
+        return (cla.Array,)
 
-    def from_numpy(self, array: Union[np.ndarray, _ScalarLike]):
+    def _map_container(self, func, array, allowed_types=None, *, strict=False):
```

Comment:
`_rec_map_container`?

Reply:
Renamed in https://github.com/inducer/arraycontext/compare/f4ab671ce9eb96e9dd2397d9a77d5e7ac7c9be36..971df793aed6b5b107be09588c0791a2f664c2b9